spotify/job/dates.py

```diff
@@ -0,0 +1,61 @@
+import datetime as dt
+from loguru import logger
+import sys
+
+
+class DateRange:
+    """
+    Represents a range of dates.
+    """
+
+    def __init__(self, start: dt.datetime, end: dt.datetime) -> None:
+        self.start = start
+        self.end = end
+
+        # Calculate the number of days between start and end date
+        self.days = (end - start).days
+
+    def __iter__(self):
+        """
+        Iterate over all days in the range
+        """
+        for i in range(self.days):
+            yield (self.start - dt.timedelta(days=i), self.end - dt.timedelta(days=i))
```

Comment:
is this correct? you move the range and also return a range, shouldn't it be just one day?

Reply:
it's not correct; will fix